src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs

```diff
@@ -213,24 +217,20 @@ protected virtual PropertyModel CreatePropertyModel(PropertyInfo propertyInfo)
                 throw new ArgumentNullException(nameof(propertyInfo));
             }
 
-            // CoreCLR returns IEnumerable<Attribute> from GetCustomAttributes - the OfType<object>
-            // is needed to so that the result of ToArray() is object
             var attributes = propertyInfo.GetCustomAttributes(inherit: true);
-            var propertyModel = new PropertyModel(propertyInfo, attributes);
-            var bindingInfo = BindingInfo.GetBindingInfo(attributes);
-            if (bindingInfo != null)
+            var propertyModel = new PropertyModel(propertyInfo, attributes)
```

Comment:
Like this cleanup. But please move the `var fredsModel` variable declarations closer to their first use. In this case, `propertyModel` isn't used 'til the method returns.